File: app/src/main/java/android/examples/retrofitcep/activity/MainActivity.java
package android.examples.retrofitcep.activity;

import android.app.ProgressDialog;
import android.examples.retrofitcep.R;
import android.examples.retrofitcep.model.Cep;
import android.examples.retrofitcep.rest.ApiInterface;
import android.examples.retrofitcep.utils.Mask;
import android.examples.retrofitcep.utils.Utils;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import retrofit.Callback;
import retrofit.RestAdapter;
import retrofit.RetrofitError;
import retrofit.client.Response;

public class MainActivity extends AppCompatActivity {

    EditText cep_user;
    Button button_user;
    TextView cep, logradouro, complemento, bairro, localidade, uf;
    ApiInterface apiInterface;
    RestAdapter restAdapter;
    ProgressDialog progressDialog;
    TextWatcher cepMask;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
        getJson();
        configMask();


    }

    private void configMask() {
        cepMask = Mask.insert("#####-###", cep_user);
        cep_user.addTextChangedListener(cepMask);
    }

    private void getJson() {

        restAdapter = new RestAdapter.Builder().setEndpoint(ApiInterface.URL).build();

        apiInterface = restAdapter.create(ApiInterface.class);

        button_user.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {

                String editTextCep = cep_user.getText().toString();

                if (editTextCep.length() <= 0) {

                    Utils.showSnack(MainActivity.this, "Preencha o CEP");

                } else {

                    getProgress();

                    apiInterface.getCep(editTextCep, new Callback<Cep>() {

                        @Override
                        public void success(Cep cepResponse, Response response) {

                            cep.setText(cepResponse.getCep());
                            logradouro.setText(cepResponse.getLogradouro());
                            complemento.setText(cepResponse.getComplemento());
                            bairro.setText(cepResponse.getBairro());
                            localidade.setText(cepResponse.getLocalidade());
                            uf.setText(cepResponse.getUf());

                            progressDialog.cancel();

                        }

                        @Override
                        public void failure(RetrofitError error) {

                            progressDialog.cancel();

                            if (error.getKind().equals(RetrofitError.Kind.NETWORK)) {
                                Utils.showSnack(MainActivity.this, "Não há conexão com a internet");
                            }


                        }
                    });
                }

                clearField();

            }


        });


    }

    private void clearField() {
        cep_user.setText("");
    }

    private ProgressDialog getProgress() {

        progressDialog = new ProgressDialog(MainActivity.this);
        progressDialog.setTitle("Consulta CEP");
        progressDialog.setMessage("Carregando Dados...");
        progressDialog.show();


        return progressDialog;
    }

    private void initViews() {
        cep = (TextView) findViewById(R.id.cep);
        logradouro = (TextView) findViewById(R.id.logradouro);
        complemento = (TextView) findViewById(R.id.complemento);
        bairro = (TextView) findViewById(R.id.bairro);
        localidade = (TextView) findViewById(R.id.localidade);
        uf = (TextView) findViewById(R.id.uf);
        cep_user = (EditText) findViewById(R.id.cep_user);
        button_user = (Button) findViewById(R.id.button_user);

    }


}

File: app/src/main/java/android/examples/retrofitcep/activity/MainActivity.kt
package android.examples.retrofitcep.activity

import android.Manifest
import android.app.ProgressDialog
import android.examples.retrofitcep.R
import android.examples.retrofitcep.rest.DataCEP
import android.examples.retrofitcep.utils.Mask
import android.examples.retrofitcep.utils.Utils
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextWatcher
import android.util.Log
import android.view.View
import io.vrinda.kotlinpermissions.PermissionCallBack
import io.vrinda.kotlinpermissions.PermissionsActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.startActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : PermissionsActivity() {

    private lateinit var progressDialog: ProgressDialog
    private var cepMask: TextWatcher? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        configToolbar()
        configMask()
    }

    override fun onResume() {
        super.onResume()

        requestPermission()
    }

    private fun requestPermission() {

        val listPermissions = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION)

        requestPermissions(listPermissions, object : PermissionCallBack {
            override fun permissionGranted() {
               // super.permissionGranted()
                Log.v("Call permissions", "Granted")
            }

            override fun permissionDenied() {
              //  super.permissionDenied()

                Log.v("Call permissions", "Denied")
            }
        })
    }

    private fun configToolbar() {
        setSupportActionBar(my_toolbar)
        supportActionBar?.title = "EasyCep"


    }

    fun buttonPesquisarCep(view: View) {

        getJson()
    }

    private fun configMask() {
        cepMask = Mask.insert("#####-###", cep_user)
        cep_user.addTextChangedListener(cepMask)
    }

    private fun getJson() {

        val callcep = Utils.apiService?.getCep(cep_user.text.toString())

        if (cep_user.text.isEmpty()) {

            Utils.showSnack(cep_user, "Preencha o CEP")

        } else {

            progressDialog()

            object : Thread() {
                override fun run() {
                    callcep?.enqueue(object : Callback<DataCEP> {
                        override fun onResponse(call: Call<DataCEP>?, response: Response<DataCEP>?) {

                            val r = response?.body()

                            if (response?.isSuccessful as Boolean) {

                                progressDialog.dismiss()

                                cep.text = r?.cep
                                logradouro.text = r?.logradouro
                                complemento.text = r?.complemento
                                bairro.text = r?.bairro
                                localidade.text = r?.localidade
                                uf.text = r?.uf


                                Utils.hideKeyboard(this@MainActivity)
                                clearField()


                            } else {

                                progressDialog.dismiss()

                                Utils.hideKeyboard(this@MainActivity)
                                Utils.showSnack(cep_user, "CEP não encontrado")
                                clearTextView()

                            }


                        }

                        override fun onFailure(call: Call<DataCEP>?, t: Throwable?) {

                        }

                    })
                }
            }.start()

        }


    }

    private fun clearTextView() {
        cep.text = ""
        logradouro.text = ""
        complemento.text = ""
        bairro.text = ""
        localidade.text = ""
        uf.text = ""
    }

    private fun clearField() {
        cep_user.setText("")
    }

    private fun progressDialog(): ProgressDialog {

        progressDialog = ProgressDialog(this@MainActivity)
        progressDialog.setTitle("Consulta CEP")
        progressDialog.setCancelable(false)
        progressDialog.setMessage("Carregando Dados...")
        progressDialog.show()


        return progressDialog

    }

    fun buttonVerNoMapa(view: View) = startActivity<MapaActivity>()


}
